packages/selfcare-v2-client/src/client.ts

```diff
@@ -1,20 +1,52 @@
 import { ZodiosInstance } from "@zodios/core";
 import { selfcareV2ClientApi } from "pagopa-interop-api-clients";
+import axios, { AxiosInstance } from "axios";
+import { z } from "zod";
 import { SelfCareConfig } from "./config/selfcareConfig.js";
 
 export type SelfcareV2InstitutionClient = ZodiosInstance<
   typeof selfcareV2ClientApi.institutionsApi.api
 >;
 
+const createAxiosInstance = (selfcareApiKey: string): AxiosInstance => {
+  const instance = axios.create({
+    headers: {
+      "Ocp-Apim-Subscription-Key": selfcareApiKey,
+    },
+  });
+
+  instance.interceptors.response.use(
+    (response) => response,
+    (error) => {
+      if (!axios.isAxiosError(error)) {
+        return Promise.reject(error);
+      }
+
+      const parsedErrorResponseData = z
+        .object({
+          statusCode: z.number(),
+          message: z.string(),
+        })
+        .safeParse(error.response?.data);
+
+      if (parsedErrorResponseData.success) {
+        const { statusCode, message } = parsedErrorResponseData.data;
+        // eslint-disable-next-line functional/immutable-data
```

Comment:
I suggest to move it on top file with others eslint-disable

Reply:
> I suggest to move it on top file with others eslint-disable

In this case I prefer to leave it inline since we don't want to abuse it, we usually put it on top only in test files